### PR TITLE
Snarkyjs: Expose constraint system as JSON

### PR DESCRIPTION
**Pull request by mitschabaude**
Exposes the constraint system as a JSON object to snarkyjs

The serialization is not optimal yet - a field element (gate coefficient) gets represented as an array of numbers (bytes)
We fix this on the JS side (https://github.com/o1-labs/snarkyjs/pull/529) by turning it into a decimal string

closes https://github.com/o1-labs/snarkyjs/issues/626

### CODE REVIEW

**Review by mimoo — APPROVED** [review]
Nice

(it'd be good to fix on the rust side eventually :D so I can use it as well)

File: src/lib/crypto/kimchi_bindings/wasm/src/lib.rs [hunk 1]
```diff
@@ -121,3 +121,44 @@ pub use {
     oracle::sponge::caml::CamlScalarChallenge,
 };
 */
+
+pub mod to_move {
```
Comment (mimoo) [inline src/lib/crypto/kimchi_bindings/wasm/src/lib.rs:125]: can you actually move this to a `circuit.rs` perhaps?

**Comment by mitschabaude** [comment]
!ci-build-me

**Review by bkase — APPROVED** [review]
!approved-for-mainnet

**Comment by mitschabaude** [comment]
!ci-build-me

**Pull request merged**